pages/1_14/getting_started/new_project.md

```diff
@@ -0,0 +1,164 @@
+every time you want to release a new version of your mod.
+
+The `group` is your base package name. Normally this will be basically the package structure of your mod between the
+"java" package and your main location where your classes are located. Taking the example mod as an example here, the
+structure is `/src/main/java/com/example/examplemod`. In this case, I'd use `com.example.examplemod` as the `group`.
+However when you rename the example mod, you'd use a name for the author (probably your own alias or team name) and the
+mod ID. For example, for the mod [Sparks Hammers](https://github.com/thebrightspark/SparksHammers) you will find the
+structure is `/src/main/java/brightspark/sparkshammers` so the `group` is set to `brightspark.sparkshammers`.
+
+The `archivesBaseName` will be used for the built JAR file name (along with the version), and ideally would be your mod
+ID.
+
+Further down the file we can find the following section:
+```groovy
+minecraft {
+    // The mappings can be changed at any time, and must be in the following format.
+    // snapshot_YYYYMMDD   Snapshot are built nightly.
+    // stable_#            Stables are built at the discretion of the MCP team.
+    // Use non-default mappings at your own risk. they may not always work.
+    // Simply re-run your setup task after changing the mappings to update your workspace.
+    mappings channel: 'snapshot', version: '20190719-1.14.3'
+```
+This part defines a few fields used when building your development environment (which we'll get to soon). 
+
+The `mappings` are an important one - these are the MCP mappings version. The mappings are used to convert the hard to
+read decompiled class, method and field names in code into much more readable names. Whenever you create a new project
+(like this right now!) or update Forge, you should really make sure this is updated to the latest mappings version.
+You can get the latest version over at the [MCP Bot exports site](http://export.mcpbot.bspk.rs/). You should always use
+the latest stable release for your Minecraft version where possible, otherwise use the latest snapshot. To update you
+simply just need to enter the version number into the value as explained in the comments above it.
+
+The rest of the `minecraft` section contains information about the run tasks gradle will generate for your IDE. You 
+generally don't need to mess with those.
+
+### Dependencies
+
+```groovy
+dependencies {
+    minecraft 'net.minecraftforge:forge:1.14.4-28.1.26'
+
+    // Other stuff
+}
+```
+
+In this section you will add any dependencies your project may have.
+At the beggining the only thing that is added here is the forge version.
+
+If you ever want to update the version of
+Forge within your mod, simply update this value.
+
+You may put jars on which you depend on in ./libs or you may define them like so:
+```groovy
+compile "some.group:artifact:version:classifier"
+compile "some.group:artifact:version"
```

Comment:
`compile` is the old way of writing gradle dependencies.
Don't use the `java` gradle plugin but instead use `java-library` (which means `api`/`implementation`/...)
More info in the [docs](https://docs.gradle.org/current/userguide/dependency_management_for_java_projects.html).

Reply:
Hm that was pulled out straight from the latest `build.gradle`

Reply:
Also i don't see the docs mention how do you define optional dependencies at all which is important i think? (Unles you just still use `provided`?)

Reply:
Because optional dependencies are a concept introduced by Forge, not Gradle. Unfortunately I don't know myself, maybe you can find an example addon mod?

Reply:
Asked in the forge discord server, let's what they'll say

Reply:
Looks like using any of the above is fine :man_shrugging: I guess i'll just have a link to the new docs + an explanation